### PR TITLE
Carminite debris effect

### LINKED ISSUE #2154: Restore 1.7.10 Carminite Reactor Visuals
1.7.10 had very different visual effects which we should look into restoring.

image

### PR DESCRIPTION
**Pull request by Sleepy-Horse**
This PR restores the carminite debris effect and returns the 3x3x3 reactor spawn.
Now, it spawns in a random box shape with a volume >1/8 of a cube. The textures are randomly selected from netherrack, nether portal, obsidian, and bedrock. The shape and textures reroll every tick with a 20% probability.
It has NBT data for destroying, rerolling, shape, etc., allowing map creators to use it.

Before:
image
image

After:
image
image

Fix #2154

### CODE REVIEW

File: src/main/java/twilightforest/block/entity/ReactorDebrisBlockEntity.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+public class ReactorDebrisBlockEntity extends BlockEntity {
+	private static final ResourceLocation[] TEXTURES = {
+		ResourceLocation.withDefaultNamespace("block/netherrack"),
+		ResourceLocation.withDefaultNamespace("block/bedrock"),
+		ResourceLocation.withDefaultNamespace("block/nether_portal"),
+		ResourceLocation.withDefaultNamespace("block/obsidian"),
+	};
+	public static final ResourceLocation DEFAULT_TEXTURE = TEXTURES[0];
+	private static final float Z_FIGHTING_MIN = 0.001F;
```
Comment (Tamaized) [inline src/main/java/twilightforest/block/entity/ReactorDebrisBlockEntity.java]: This needs to be in intervals of 0.008 due to certain.. performance.. mods fucking with the vertex buffer

File: src/main/java/twilightforest/client/renderer/tileentity/ReactorDebrisRenderer.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+		float maxZ = entity.maxPos.z;
+
+		renderSide(info, getSprite(entity.textures[0]), Axis.X, minX, minY, minZ, maxY, maxZ, -1);
+		renderSide(info, getSprite(entity.textures[1]), Axis.X, maxX, minY, maxZ, maxY, minZ, 1);
+		renderSide(info, getSprite(entity.textures[2]), Axis.Y, minY, minX, maxZ, maxX, minZ, -1);
+		renderSide(info, getSprite(entity.textures[3]), Axis.Y, maxY, minX, minZ, maxX, maxZ, 1);
+		renderSide(info, getSprite(entity.textures[4]), Axis.Z, minZ, minX, minY, maxX, maxY, -1);
+		renderSide(info, getSprite(entity.textures[5]), Axis.Z, maxZ, maxX, minY, minX, maxY, 1);
+
+		// Duplication for inner side because portal is transparent
```
Comment (Tamaized) [inline src/main/java/twilightforest/client/renderer/tileentity/ReactorDebrisRenderer.java]: Would it be better to disable culling instead?

**Review by Tamaized — CHANGES_REQUESTED** [review]

File: src/main/java/twilightforest/client/renderer/tileentity/ReactorDebrisRenderer.java (hunk 2 above)
Comment (Sleepy-Horse) [reply src/main/java/twilightforest/client/renderer/tileentity/ReactorDebrisRenderer.java]: Then the nether portal texture won't be transparent. Are you sure?
Before:
image

After:
image

Comment (Tamaized) [reply src/main/java/twilightforest/client/renderer/tileentity/ReactorDebrisRenderer.java]: Oh yeah I'm dumb, it didn't click in my head it'd be rendering twice. Yeah what you have is fine.

**Review by Tamaized — APPROVED** [review]

**Pull request merged**